### PR TITLE
[ROCKETMQ-89] WS_DOMAIN_NAME, SUBGROUP default values overrides custom values 

### PR DESCRIPTION
**Pull request by lizhanhui**
WS_DOMAIN_NAME, SUBGROUP default values override custom values passed by java options options

### CODE REVIEW

**Comment by vongosling** [comment]
All checks have failed @lizhanhui

**Comment by lizhanhui** [comment]
We need to disable reuse forks for the broker module as the newly added unit test requires a more isolated environment to run.

**Comment by lizhanhui** [comment]
> [ERROR] Failed to execute goal org.eluder.coveralls:coveralls-maven-plugin:4.3.0:report (default-cli) on project rocketmq-all: Processing of input or output data failed: Report submission to Coveralls API failed with HTTP status 422: Unprocessable Entity (Couldn't find a repository matching this job.) -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException


Is the service so unreliable?

File: broker/pom.xml [hunk 1]
```diff
@@ -66,4 +66,17 @@
             <artifactId>javassist</artifactId>
         </dependency>
     </dependencies>
+
+    <build>
+        <plugins>
+            <plugin>
+                <artifactId>maven-surefire-plugin</artifactId>
```
Comment (zhouxinyu) [inline broker/pom.xml]: It seems that `maven-surefire-plugin` exists in parent pom.xml?

Comment (lizhanhui) [inline broker/pom.xml]: Yes, the key point here is to override <reuseForks>false</reuseForks>.

**Comment by lizhanhui** [comment]
@shroman @zhouxinyu @vongosling Any idea on this issue?

**Comment by lizhanhui** [comment]
> But I don't quite understand why it has to have so many levels -- why not to remove rmqAddressServerDomain and let the user use only rocketmq.namesrv.domain ... ?

Yes, I am also considering centralizing default values to an external config-default.conf file and expose config.conf for user to override. This is best conducted in a dedicated PR after discussion.
For this PR, let's fix the logical fault first.

**Comment by lizhanhui** [comment]
Merged.

**Pull request closed without merging**